src/index.js

```diff
@@ -9,11 +9,17 @@ function TestHook({ callback, hookProps, children }) {
 function renderHook(callback, { initialProps, ...options } = {}) {
   const result = { current: null }
   const hookProps = { current: initialProps }
+  const resolvers = []
+  const nextUpdate = () =>
+    new Promise((resolve) => {
+      resolvers.push(resolve)
+    })
 
   const toRender = () => (
     <TestHook callback={callback} hookProps={hookProps.current}>
       {(res) => {
         result.current = res
+        resolvers.splice(0, resolvers.length).forEach((resolve) => resolve())
```

Comment:
What if we pass res to resolve? Like this `resolvers.splice(0, resolvers.length).forEach((resolve) => resolve(res))`
Would it allow us to rename nextUpdate to nextValue and do assertions directly on it?
Like `expect(async nextValue()).toEqual('Bob')`

Reply:
I did start out with that, but decided that, as there is already a way of accessing the value, it might be confusing to use different variables before and after the `await` fold

```js
test('something', async () => {
  const { result, nextValue } = renderHook(() => useSomeHook())

  expect(result.current).toBe({/* ... */})

  expect(await nextValue()).toBe({/* ... */})
})
```

It also falls down when you want to assert a nexted value

```js
test('something', async () => {
  const { result, nextValue } = renderHook(() => useSomeHook())

  expect(result.current.someProp).toBe({/* ... */})

  // doesn't work
  // expect(await nextValue().someProp).toBe({/* ... */})

  const next = await nextValue()
  expect(next.someProp).toBe({/* ... */})
})
```

or multiple assertions

```js
test('something', async () => {
  const { result, nextValue } = renderHook(() => useSomeHook())

  expect(result.current).not.toContain({/* ... */})
  expect(result.current).toHaveLength(2)

  // fine
  // expect(await nextValue()).toContain({/* ... */})
  // hangs until the test fails
  // expect(await nextValue()).toHaveLength(3)

  const next = await nextValue()
  expect(next).toContain({/* ... */})
  expect(next).toHaveLength(3)
})
```

It also raises questions about whether it should fire if the `result.current` is set to the same value as it was already set to.  Consider a hook that has some internal state that updated asyncronously, triggering a side effect , but the hook returns a static result or no result at all itself (e.g. a `useRedux` hook that returns the static store, but a dispatch triggers some internal state to be updated).  If it didnt trigger on every render, this would once again be harder to test.

If we wanted to go this way, I think `result.nextValue()` _feels_ better to me for what it represents, but right now I still feel as the `nextUpdate` approach in this PR is more consistent, cleaner in more (common, in my opinion) use cases and requires fewer additional variables in complex tests.

Reply:
I do agree with you. Anyway, we can access the current right after we awaited so it doesn't really matter. 

About the method's name, I am wondering if it wouldn't be better to have something like the async methods in react-testing-library.

I am talking about the wait, waitForElement and waitForDomChange methods.

Maybe a name like waitForNextUpdate or waitForNextResult would provide a better description?

Reply:
> waitForNextUpdate

I like that. `waitForNextResult` is fine but there are legitimate cases where a custom may not have a result at all (i.e. no return value), but `waitForNextUpdate` covers both nicely.

@ab18556 if you want to submit a PR [adding yourself as a contributor](https://github.com/mpeyper/react-hooks-testing-library/blob/master/CONTRIBUTING.md#add-yourself-as-a-contributor) (`review`, `ideas`, others?), I'll happily merge it.

Reply:
I don't fell i did that much, but since it was my first contribution to a public project, I am happy to do it.

Reply:
You had more input on this PR than anyone else, and as a library author and maintainer, having anyone actually challenging and validating your work is invaluable and a fantastic contribution you can make (second only to updating the documentation 😉).